R/Seurat_III.R

```diff
@@ -512,17 +525,23 @@ RemoveCellCycle <- function(seuratObj, runPCAonVariableGenes = F, pcaResultFile
     seuratObj <- RunPCA(object = seuratObj, pc.genes = VariableFeatures(object = seuratObj), do.print = F, verbose = F)
   }
 
-  #Eisa: is this really correct to repeat regression here?
-  if (!useSCTransform) {
-    seuratObj <- ScaleData(object = seuratObj, features = rownames(x = seuratObj), vars.to.regress = c("nCount_RNA", "p.mito"), display.progress = F, verbose = F)
-  } else {
-    seuratObj <- SCTransform(seuratObj, vars.to.regress = c("nCount_RNA", "p.mito"), verbose = FALSE)
-  }
+  # #Eisa: is this really correct to repeat regression here?
+  # # EM: I added features = rownames(x = seuratObj) above and that is all we need i belive
+  # if (!useSCTransform) {
+  #   seuratObj <- ScaleData(object = seuratObj, features = rownames(x = seuratObj), vars.to.regress = c("nCount_RNA", "p.mito"), display.progress = F, verbose = F)
+  # } else {
+  #   seuratObj <- SCTransform(seuratObj, vars.to.regress = c("nCount_RNA", "p.mito"), verbose = FALSE)
+  # }
 
   if (!is.null(pcaResultFile)) {
     SeuratObjsCCPCA$CellBarcode <- colnames(seuratObj)
     write.table(SeuratObjsCCPCA, file = pcaResultFile, sep = '\t', row.names = F, quote = F)
   }
+  
+  # print("Scaling data with full set of genes")
```

Comment:
why commented out?

Reply:
Not sure. Maybe it is a legacy. Will fix.

Reply:
it is definitely new (or at least moved), and doesnt it make sense to do?